File: modulefederation/libs/portal/src/lib/tracer/traces/console-trace.ts
import { Trace } from '../trace';
import { TraceEntry } from '../trace-entry';
import { TraceFormatter } from '../trace-formatter';

/**
 * A <code>ConsoleTrace</code> will emit trace entries to the console.
 */
export class ConsoleTrace extends Trace {
    // constructor

    constructor(messageFormat : string) {
        super(new TraceFormatter(messageFormat));
    }

    // implement Trace

    /**
     * @inheritdoc
     */
    trace(entry : TraceEntry) : void {
        console.log(this.format(entry));
    }
}

File: common/src/main/kotlin/org/sirius/common/tracer/trace/ConsoleTrace.kt
package org.sirius.common.tracer.trace
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import org.sirius.common.tracer.Trace
import org.sirius.common.tracer.TraceEntry
import org.sirius.common.tracer.TraceFormatter

class ConsoleTrace : Trace() {
    // override

    override fun trace(entry: TraceEntry, format: TraceFormatter) {
        println(format.format(entry))
    }
}